05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+       1*y = 0
+\]
+Now we can write this as a matrix calculation.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    = 0  
+\]
+
+We have now encoded this simply rule as a matrix product. An agent can input their versions of x and y, and perform the calculation.
+The agent can then check if their result is 0, if so they are compliant and can pay $y$ tax. Let us revisit our rule, and evolve it a little.
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x"
+\linebreak
+\linebreak
+We can rewrite the matrix equation as an inequality.
+\[
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    >= 0  
+\]
+The addition of the "at least" adds a little bit of complexity to our matrix schema. In that we may not be able to simply check for equality to a constant, and may instead need to check for a certain condition.
+To make these comparisons easier, we have decided to ensure that whatever condition, equality or inequality, we would compare to 0.
+However if our tax rule looked like this (for whatever reason):
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x + 5"
+\linebreak
+\linebreak
+Then it would be impossible to capture that using the current matrix calculation, so to capture these cases we introduce a constant to the input list, as below.
+\[
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y \\
+        1
+    \end{bmatrix}   
+    >= 0  
+\]
+We have now managed to capture any linear equation or inequality with any constant shift. This is where we decided to limit the scope of our rules from a technical aspect.
+The final section of our schema is how to inform an agent whether it should be looking for equality or inequality.
+For this, we package with the rule-matrix an "auxiliary" vector containing a list of comparison codes.
+\begin{table}[h]
+    \centering
+    \begin{tabular}{ll}
+    \hline
+    \multicolumn{1}{|l}{Aux code} & \multicolumn{1}{l|}{Meaning} \\ \hline
+    0                             & = 0                          \\
+    1                             & \textgreater 0               \\
+    2                             & \textgreater{}= 0            \\
+    3                             & != 0                         \\
+    4                             & real                        
+    \end{tabular}
+\end{table}
+\linebreak
+The aux code 4 is reserved for when we want the raw result of the matrix multiplication, which is used sparingly for certain rules.
+We have now arrived at the representation of rules we are using for our project. For the rule above:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
```

Comment:
go to ANOTHER equation.. do NOT put them side by side plzzzz

`\begin{equation} \end{equation}` should do the work :)